### PR TITLE
docs: clarify behavior of --extras and subcommands

### PR DESCRIPTION
**Pull request by evanrittenhouse**
# Pull Request Check List

Resolves: https://github.com/python-poetry/poetry/issues/7556

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

Documentation only

### CODE REVIEW

File: docs/pyproject.md [hunk 1]
```diff
@@ -395,6 +395,12 @@ You can install all extras with the `--all-extras` option:
 poetry install --all-extras
 ```
 
+Note that `install --extras` and the variations mentioned above (`--all-extras`, `--extras foo`, etc.) only work on dependencies defined in the current project. If you want to install extras defined by dependencies, you'll have to express that in the dependency itself:
+```toml
+[tool.poetry.group.dev.dependencies]
+fastapi = {version="^0.92.0", extras=["all"]}
+```
+
```
Comment (PabloEmidio) [inline docs/pyproject.md:403]: **Would not be better putting that note as a `note` using `{{% note %}}`?**

```suggestion
{{% note %}}
Note that `install --extras` and the variations mentioned above (`--all-extras`, `--extras foo`, etc.) only work on dependencies defined in the current project. If you want to install extras defined by dependencies, you'll have to express that in the dependency itself:
```toml
[tool.poetry.group.dev.dependencies]
fastapi = {version="^0.92.0", extras=["all"]}
...
{{% /note %}}
```

Comment (evanrittenhouse) [reply docs/pyproject.md]: Done in latest commit - good suggestion, thanks @PabloEmidio

**Review by PabloEmidio — APPROVED** [review]
## Appear to be all set

image

**Comment by evanrittenhouse** [comment]
Re-requesting review since I had to rebase against master

**Review by PabloEmidio — APPROVED** [review]

**Comment by evanrittenhouse** [comment]
Looks like it still needs a review from someone with write access - @finswimmer , possible to help out here?

**Review by radoering — APPROVED** [review]

**Pull request merged**